### DISCUSSION TITLE
Get one single view given multiple system of records

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#115] [Q&A] [answered]
Hi guys,
I'm in a situation in which I have two system of records with related projections, and the output I want is to get one single view with aggregation from data retrieved by all the projections.

Reading the documentation (https://docs.mia-platform.eu/docs/fast_data/configuration/single_views#create-the-single-view-creator-service) from the picture I understand that for each system of record I associate a single_view_creator that will generate the single view I want.

The first question is: to introduce multiple sv-creators for the same single view, wouldn't introduce duplication of aggregation logic and also, a big chance to introduce errors?

The second question is: what is the best practice in naming these sv-creators? For example let's say I'm creating the single view of the user, and I'm getting data from both SoR_A and SoR_B, I would have `user-A-single-view-creator` and `user-B-single-view-creator`?

Thanks!

**Comment by fedepini** [comment, marked as answer]
Hi @elmalakomar,

we usually suggest having one Single View Creator and using multiple Real Time Updaters to write on the same Projection Changes collection, but your case is different.

As for the first question, multiple Single View Creators shouldn't generate errors because the different Single View Creators are working on the same DB, which is kept updated by the different Real Time Updaters.

As for the naming convention, we don't have one for the multiple Single View Creator case, while we suggest this format if only one Single View Creator is used: `fast-data-{{single-view-name}}-single-view-creator`.
